packages/stark-ui/src/modules/keyboard-directives/directives/transform-input.directive.spec.ts

```diff
@@ -0,0 +1,111 @@
+ * @param element - the HTMLInputElement to mock the event on
+ */
+const mockInputEvent: (value: string, element: HTMLInputElement) => void = (value: string, element: HTMLInputElement): void => {
+	const event: Event = document.createEvent("Event");
+	event.initEvent("input", true, true);
+	element.value = value;
+	element.dispatchEvent(event);
+};
+
+describe("TransformsInputDirective with ngModel", () => {
+	/**
+	 * @ignore
+	 * mocked wrapper for the directive
+	 */
+	@Component({
+		selector: "test-component",
+		template: "<input [(ngModel)]='value' starkTransformInput='upper'/>"
+	})
+	class TestComponent {
+		public value: string = "";
+	}
+
+	let fixture: ComponentFixture<TestComponent>;
+
+	beforeEach(() => {
+		TestBed.configureTestingModule({
+			declarations: [StarkTransformInputDirective, TestComponent],
+			imports: [FormsModule],
+			providers: [{ provide: STARK_LOGGING_SERVICE, useValue: new MockStarkLoggingService() }]
+		});
+
+		fixture = TestBed.createComponent(TestComponent);
+		// trigger initial data binding
+		fixture.detectChanges();
+	});
+
+	it("should set the value to uppercase", () => {
+		expect(fixture.componentInstance.value).toBe("", "field 'value' should start as empty string ");
+
+		const inputElement: HTMLInputElement = <HTMLInputElement>fixture.debugElement.query(By.css("input")).nativeElement;
+		mockInputEvent("a", inputElement);
+		fixture.detectChanges();
+
+		expect(fixture.componentInstance.value).toBe("A", "field 'value' should have been updated to upper case 'A'");
+	});
+});
+
+describe("TransformsInputDirective with formControl", () => {
+	/**
+	 * @ignore
+	 * mocked wrapper for the directive
+	 */
+	@Component({
+		selector: "test-component",
+		template: "<input [formControl]='formControl' starkTransformInput='lower'/>"
+	})
+	class TestComponent implements OnInit {
+		public formControl: FormControl = new FormControl("");
+
+		public ngOnInit(): void {
+			this.formControl.valueChanges.subscribe(this.onValueChanges);
+		}
+
+		/**
+		 * @ignore
+		 * Placeholder for spy
+		 */
+		public onValueChanges(): void {
+			/*noop*/
+		}
+	}
+
+	let fixture: ComponentFixture<TestComponent>;
+
+	beforeEach(() => {
+		TestBed.configureTestingModule({
+			declarations: [StarkTransformInputDirective, TestComponent],
+			imports: [ReactiveFormsModule],
+			providers: [{ provide: STARK_LOGGING_SERVICE, useValue: new MockStarkLoggingService() }]
+		});
+
+		fixture = TestBed.createComponent(TestComponent);
+		spyOn(fixture.componentInstance, "onValueChanges").calls.reset();
+
+		// trigger initial data binding
+		fixture.detectChanges();
+	});
+
+	it("should set the value to uppercase", () => {
+		expect(fixture.componentInstance.formControl.value).toBe("", "FormControl value should start as empty string ");
+
+		const inputElement: HTMLInputElement = <HTMLInputElement>fixture.debugElement.query(By.css("input")).nativeElement;
+		mockInputEvent("A", inputElement);
```

Comment:
You can also mock typing a non-word character and also an lowercase char to test that the chars remains "as is'... you know... just to be sure ;)